sugarloaf/src/sugarloaf.rs

```diff
@@ -764,29 +768,23 @@ impl Sugarloaf {
                         .render_with_encoder(0, view, &mut encoder, None);
                 }
 
+                self.layer_brush.end_frame();
+
                 self.rect_brush
                     .render(&mut encoder, view, &self.rects, &mut self.ctx);
-
                 self.rects = vec![];
                 self.current_row = 0;
 
                 let _ = self
                     .text_brush
                     .draw_queued(&mut self.ctx, &mut encoder, view);
 
-                self.layer_brush.end_frame();
-
                 for entry_render in
```

Comment:
I am aware it isn't part of your change, but isn't the clone redundant since you removed the logs that used `entry`. 

```rust
for entry in self.graphic_rects.values() {
    if let Some(graphic_data) = self.graphics.get(&entry.id) {
        let rows = entry.end_row - entry.start_row;
        let height = (rows - 2.) * self.layout.scaled_sugarheight;

        let a = layer::types::Image::Raster {
            handle: graphic_data.handle.clone(),
            bounds: Rectangle {
                x: entry.pos_x,
                y: entry.pos_y,
                width: entry.width as f32,
                height,
            },
        };

        self.layer_brush
            .prepare_ref(&mut encoder, &mut self.ctx, &[&a]);

        self.layer_brush
            .render_with_encoder(0, view, &mut encoder, None);
    }
}
```

Reply:
frankly I don't remember (has been a while that I have touched the graphic media), the work with layer_brush to render images is still incomplete so left the way it's 😢 . Do you have any experience with sixel? I would like to finish though in another PR.

Reply:
I do not. I was merely curious 😄

Reply:
haha fair enough 👌